Lib/typing.py

```diff
@@ -1429,18 +1430,31 @@ def name_by_id(user_id: UserId) -> str:
         num = UserId(5) + 1     # type: int
     """
 
-    def new_type(x):
-        return x
+    __slots__ = ('__name__', '__qualname__', '__supertype__')
+
+    def __init__(self, name, tp):
+        self.__name__ = self.__qualname__ = name
+        self.__supertype__ = tp
+
+    @staticmethod
+    def __call__(arg):
+        return arg
 
-    new_type.__name__ = name
-    new_type.__supertype__ = tp
-    return new_type
+    def __repr__(self):
+        """ NewType reprs are in the form:
+           “NewTClassName<typename:supertypename>”, e.g.:
```

Comment:
Please don't use non-ASCII quotes. Also I am not sure this docstirng is actually needed, the code is self-explanatory.